app/models/metric/ci_mixin/capture.rb

```diff
@@ -17,6 +17,18 @@ def queue_name_for_metrics_collection
     ems.metrics_collector_queue_name
   end
 
+  def split_capture_intervals(start_time, end_time, threshold=1.day)
+    raise _("Start time must be earlier than End time") if start_time > end_time
+    # Create an array of ordered pairs from start_time and end_time so that each ordered pair is contained
+    # within the threshold.  Then, reverse it so the newest ordered pair is first:
+    # start_time = 2017/01/01 12:00:00, end_time = 2017/01/04 12:00:00
+    # [[2017-01-03 12:00:00 UTC, 2017-01-04 12:00:00 UTC],
+    #  [2017-01-02 12:00:00 UTC, 2017-01-03 12:00:00 UTC],
+    #  [2017-01-01 12:00:00 UTC, 2017-01-02 12:00:00 UTC]]
+    (start_time..end_time).step_value(threshold).each_cons(2).collect { |s_time, e_time| [s_time, e_time] }.reverse
```

Comment:
When I tried it, `each_cons` returned a single array:

```ruby
> (1..20).step_value(2).each_cons(2)
=> #<Enumerator: [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 20]:each_cons(2)>
```

Whereas the `collect` will group them into the tuples:

```ruby
> (1..20).step_value(2).each_cons(2).collect {|s, e| [s, e]}
=> [[1, 3], [3, 5], [5, 7], [7, 9], [9, 11], [11, 13], [13, 15], [15, 17], [17, 19], [19, 20]]
```

And, reverse gets the ordering correct:

```ruby
> (1..20).step_value(2).each_cons(2).collect {|s, e| [s, e]}.reverse
=> [[19, 20], [17, 19], [15, 17], [13, 15], [11, 13], [9, 11], [7, 9], [5, 7], [3, 5], [1, 3]]
```